### PR TITLE
add suggestion to use `describe` to obtain container names

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:

``` release-note
release-note-none
```

Commands with the `--container=` option provide no suggestions to a user
on how to obtain a container's name from a pod.

This patch adds a suggestion on the usage output to use the `describe`
command on a pod to obtain the container value that is passed to the
`--container=` flag.

`$ kubectl exec -h`

```
Execute a command in a container.

Examples:

kubectl exec 123456-7890 date

kubectl exec 123456-7890 -c ruby-container date

kubectl exec 123456-7890 -c ruby-container -i -t -- bash -il

Flags:
  -c, --container string   Container name. If omitted, the first container in the pod will be chosen. Use the 'describe' command to find the list of containers on a pod.
  -p, --pod string         Pod name
  -i, --stdin              Pass stdin to the container
  -t, --tty                Stdin is a TTY

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30717)
<!-- Reviewable:end -->
```

### CODE REVIEW

**Comment by deads2k** [comment]
Usability isn't my area of expertise, but you can run these command against things that aren't pods.  Why would we give advice that is only valid for a pod?  Also, that's a super long description.

I'm not sold on doing this.

@juanvallejo  @fabianofranz care to elaborate on reasoning?  The help string as provided is super long, its incorrect, and it seems like most people who have multiple containers in a pod where the first one isn't the one they want would know what they have.

**Comment by juanvallejo** [comment]
@deads2k I updated the commit to match the one in https://github.com/openshift/origin/pull/10469. PTAL

File: pkg/kubectl/cmd/exec.go [hunk 1]
```diff
@@ -254,7 +261,8 @@ func (p *ExecOptions) Run() error {
 
 	containerName := p.ContainerName
 	if len(containerName) == 0 {
-		glog.V(4).Infof("defaulting container name to %s", pod.Spec.Containers[0].Name)
+		msg := fmt.Errorf("defaulting container name to %s, use '%s describe po/%s' cmd to see all containers in this pod", pod.Spec.Containers[0].Name, p.FullCmdName, p.PodName)
```
Comment (deads2k) [inline pkg/kubectl/cmd/exec.go]: Detect whether there's only one container.  If there's only one container, there's no reason to drop this message.

Comment (deads2k) [inline pkg/kubectl/cmd/exec.go]: `fmt.Fprintf(options.ErrOut, ****`?

**Comment by juanvallejo** [comment]
@deads2k 
> Detect whether there's only one container. If there's only one container, there's no reason to drop this message.

Done! Thanks for the feedback

File: pkg/kubectl/cmd/exec.go [hunk 2]
```diff
@@ -254,7 +261,10 @@ func (p *ExecOptions) Run() error {
 
 	containerName := p.ContainerName
 	if len(containerName) == 0 {
-		glog.V(4).Infof("defaulting container name to %s", pod.Spec.Containers[0].Name)
+		if len(pod.Spec.Containers) > 1 {
+			msg := fmt.Sprintf("defaulting container name to %s, use '%s describe po/%s' cmd to see all containers in this pod", pod.Spec.Containers[0].Name, p.FullCmdName, p.PodName)
+			fmt.Fprintf(p.Err, "%s", msg)
```
Comment (deads2k) [inline pkg/kubectl/cmd/exec.go]: no reason for two lines.  `fmt.Fprintf` can do both at once.

**Comment by deads2k** [comment]
nit, squash, then lgtm.

**Comment by deads2k** [comment]
lgtm

**Comment by deads2k** [comment]
bumping from 1.4.  This isn't important enough to merge during slush.

File: pkg/kubectl/cmd/exec.go [hunk 3]
```diff
@@ -254,7 +259,9 @@ func (p *ExecOptions) Run() error {
 
 	containerName := p.ContainerName
 	if len(containerName) == 0 {
-		glog.V(4).Infof("defaulting container name to %s", pod.Spec.Containers[0].Name)
+		if len(pod.Spec.Containers) > 1 {
+			fmt.Fprintf(p.Err, "defaulting container name to %s, use '%s describe po/%s' to see all containers in this pod.\n", pod.Spec.Containers[0].Name, p.FullCmdName, p.PodName)
```
Comment (liggitt) [inline pkg/kubectl/cmd/exec.go]: `describe pod/%s`? I find the letter saved in `po` more confusing than convenient

Comment (liggitt) [inline pkg/kubectl/cmd/exec.go]: `... to see all container names\n"`? punctuating like a sentence when we're not capitalizing looks weird

**Pull request merged**